openapi.yml

```diff
@@ -0,0 +1,576 @@
+      summary: Get count of threads for a user
+      description: Return count of threads for a user depending on a number of filters
+      parameters:
+        - name: user_id
+          in: query
+          description: User UUID assigned to the theads
+          schema:
+            type: string
+            format: uuid
+            example: 6934f0c7-b595-435b-aa8f-291d6b6070d3
+        - name: count_type
+          in: query
+          description: What kind of count to perform on the threads
+          schema:
+            type: string
+            enum:
+              - all
+              - unread
+              - survey
+            default: unread
+        - name: is_closed
+          in: query
+          description: True | False is the thread closed
+          schema:
+            type: bool
+            default: false
+        - name: survey
+          in: query
+          description: UUID of survey that the threads pertain to
+          schema:
+            type: string
+            format: uuid
+            example: 85e75391-9b91-4322-8984-3d53da29186e
+        - name: ru_ref
+          in: query
+          description: Business reference
+          schema:
+            type: string
+            example: 41104403458
+        - name: category
+          in: query
+          description: A string referring to what type a message is
+          schema:
+            type: string
+            enum:
+              - SURVEY
+              - TECHNICAL
+              - MISC
+            default: SURVEY
+        - name: case_id
+          in: query
+          description: UUID referring to the case
+          schema:
+            type: string
+            format: uuid
+            example: 6934f0c7-b595-435b-aa8f-291d6b6070d3
+        - name: assigned_internal_user_id
+          in: query
+          description: UUID referring to the assigned internal (ONS) user id
+          schema:
+            type: string
+            format: uuid
+            example: ca0f0819-8542-4410-bd38-edf67c59931f
+        - name: respondent_id
+          in: query
+          description: UUID referring to the id of the respondent linked to the message
+          schema:
+            type: string
+            format: uuid
+            example: ca0f0819-8542-4410-bd38-edf67c59931f
+      responses:
+        200:
+          description: Successfully returned unread thread count
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: '#/components/schemas/basicCount'
+                  - $ref: '#/components/schemas/complexCount'
+                examples:
+                  basicCount:
+                    value:
+                      total: 3
+                  complexCount:
+                    value:
+                      totals:
+                        open: 0
+                        closed: 2
+                        my_conversations: 1
+                        initial_conversations: 3
```

Comment:
We will not have a `count_type` so `basicCount` and `complexCount` is redundant. We should return these in another way. The default response (to be used in Rops requests) will look like the complex count, and the basic count should be returned in frontstage requests.